avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -57,4 +60,18 @@ protected PropertyName _findName(Annotated a)
         AvroName ann = _findAnnotation(a, AvroName.class);
         return (ann == null) ? null : PropertyName.construct(ann.value());
     }
+
+    @Override
+    public Boolean hasRequiredMarker(AnnotatedMember m) {
+        if (_hasAnnotation(m, Nullable.class)) {
+            return false;
+        }
+        // Appears to be a bug in POJOPropertyBuilder.getMetadata()
+        // Can't specify a default unless property is known to be required or not, or we get an NPE
+        // If we have a default but no annotations indicating required or not, assume required.
+        if (_hasAnnotation(m, AvroDefault.class) && !_hasAnnotation(m, JsonProperty.class)) {
```

Comment:
I don't think check for `@JsonProperty` should be needed here.
Is that only used to avoid `NPE` or for some other need?